esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -560,6 +561,30 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         return configureLogTransform(bindingId, logTransformName, configTree, overwrite, message);
     }
 
+    void preserveBindingContent(IPTree* src, IPTree* dst, IPTree* exclusion) const
+    {
+        if (src)
+        {
+            Owned<IPTreeIterator> children(src->getElements("*"));
+
+            ForEach(*children)
+            {
+                auto child = &children->query();
+
+                if (child != exclusion)
+                    dst->addPropTree(child->queryName(), LINK(child));
+            }
+
+            Owned<IAttributeIterator> attributes(src->getAttributes());
+
+            ForEach(*attributes)
+            {
+                if (!dst->hasProp(attributes->queryName()))
```

Comment:
Not sure of the exact scenarios, but just checking that the changed values shouldn't win?

Reply:
I would eventually like to be able to get a binding from one service and republish it to another without having to strip Binding and Definition attributes from the configuration first. This was a way to let the new values win. The destination node is assumed to have been assigned attributes that identify the new binding and, in the event that the source node contains potentially stale values of those attributes, the destination node's values need to win.

An alternative could be to rearrange the code that calls this to only add the new attributes after, overwriting anything preserved here.